docs/zkapps/tutorials/07-oracle.mdx

```diff
@@ -24,61 +24,76 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
+    ```sh
+    $ npm install -g zkapp-cli
+    ```
 
-Although we’re using a mock credit score API as our data source, you can easily swap out the oracle’s HTTP request to use any REST API or GraphQL API source of your choice, to create an oracle for any type of data.
+- Ensure your environment meets the [Prerequisites](/zkapps/tutorials#prerequisites) for zkApp Developer Tutorials.
 
-## Oracle
+This tutorial has been tested with:
 
-### How it works
+- [zkApp CLI](https://www.npmjs.com/package/zkapp-cli) version 0.13.0
+- [o1js](https://www.npmjs.com/package/o1js) version 0.13.0
 
-Let’s start by understanding how an oracle works.
+## High-Level Overview
 
-Mina smart contracts run off-chain and make it possible to prove that the expected computation was run on private data without revealing the data itself. But if the smart contract consumes data from a 3rd party source, we need a way to verify that this data is authentic–i.e. that it was provided by the expected source.
+In this tutorial, you build an oracle that retrieves data from a REST API and write a smart contract that consumes information from this oracle.
 
-In the future, Mina’s zkOracles will allow a zkApp to consume data trustlessly from any HTTPS data source. But for now, the oracle design described in this tutorial will typically be operated by the zkApp developer, it will fetch and sign the desired data, and then a zkApp can consume this data and will verify the signature to ensure that the data was provided by the expected source.
+1. Retrieve data from a REST API that provides mock credit score information for two users: one with a high credit score and one with a low credit score. 
+1. The smart contract consumes this information and allows users to prove their credit score is above a certain threshold (for example, higher than 700).
 
-:::note
+Using the smart contract, an end user can generate an attestation that their credit score is above a certain value. To maintain their privacy, the user can prove this fact to a third party without sharing the exact credit score or other personal information.
 
-Data providers can also operate response signers (like the one described above) to provide users with an oracle that does not require them to trust an intermediary. In other words, if a credit score or other data provider chooses to sign response data themselves, users can consume data from that source without trusting anybody besides the data provider (who users are already trusting to provide correct data).
+This tutorial uses a mock credit score API as the data source and provides a foundation to create an oracle for any type of data. Just swap out the oracle's HTTP request to use any REST API or GraphQL API source.  
```

Comment:
```suggestion
This tutorial uses a mock credit score API as the data source and provides a foundation to create an oracle for any type of data. Just alter to code to query data from whatever source you need, a REST API, for example. 
```
The existing oracle code doesn't actually make an HTTP request to another source; the values are simply hard-coded.